inst/python/rpytools/output.py

```diff
@@ -143,15 +143,16 @@ def flush(self):
     return None
 
 
-def remap_output_streams(r_stdout, r_stderr, tty, force):
-  
-  if (force or sys.stdout is None):
-    sys.stdout = OutputRemap(sys.stdout, r_stdout, tty)
-    
-  if (force or sys.stderr is None):
-    sys.stderr = OutputRemap(sys.stderr, r_stderr, tty)
-
+def remap_output_streams(r_stdout, r_stderr, tty):
+  sys.stdout = OutputRemap(sys.stdout, r_stdout, tty)
+  sys.stderr = OutputRemap(sys.stderr, r_stderr, tty)
 
+def reset_output_streams():
+  # https://stackoverflow.com/a/51340381/3297472 suggests that the initial stdout
+  # is always available in `sys.__stdout__` so we just need to reassign to revert
+  # it.
+  sys.stdout = sys.__stdout__
```

Comment:
I think it'd be safer to cache the previous streams, and then restore them. (So we don't incidentally undo any patching done by user or framework code). A context manager seems appropriate here.

Reply:
Done. I used a context manager, even though we have to call `__enter__` and `__exit__` manually.